### PR TITLE
Advancement new stuff and refactor old bad names

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/advancement/Advancement.mapping [hunk 1]
```diff
@@ -70,7 +72,9 @@ CLASS net/minecraft/class_161 net/minecraft/advancement/Advancement
 			ARG 8 hidden
 		METHOD method_698 toJson ()Lcom/google/gson/JsonObject;
 		METHOD method_699 toPacket (Lnet/minecraft/class_2540;)V
+			ARG 1 buf
 		METHOD method_700 findParent (Ljava/util/function/Function;)Z
```
Comment (Runemoro) [inline mappings/net/minecraft/advancement/Advancement.mapping:76]: Maybe this should just be `getParent`?

File: mappings/net/minecraft/predicate/entity/EntityPredicate.mapping [hunk 2]
```diff
@@ -75,3 +78,38 @@ CLASS net/minecraft/class_2048 net/minecraft/predicate/entity/EntityPredicate
...
+	CLASS class_5258 Extended
+		FIELD field_24388 EMPTY Lnet/minecraft/class_2048$class_5258;
+		FIELD field_24389 conditions [Lnet/minecraft/class_4570;
+		FIELD field_24390 combinedCondition Ljava/util/function/Predicate;
+		METHOD <init> ([Lnet/minecraft/class_4570;)V
+			ARG 1 conditions
+		METHOD method_27803 ofLegacy (Lnet/minecraft/class_2048;)Lnet/minecraft/class_2048$class_5258;
+			ARG 0 predicate
+		METHOD method_27804 toJson (Lnet/minecraft/class_5267;)Lcom/google/gson/JsonElement;
+			ARG 1 saver
```
Comment (Runemoro) [inline mappings/net/minecraft/predicate/entity/EntityPredicate.mapping]: Shouldn't `saver` be `serializer`? Saver sounds like it's writing the json to the file or something, but the method returns a JsonElement.

File: mappings/net/minecraft/advancement/Advancement.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/advancement/Advancement.mapping:76]: it is trying to find the parent and returns a boolean indicating if the parent is found.

File: mappings/net/minecraft/predicate/entity/EntityPredicate.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/predicate/entity/EntityPredicate.mapping]: the saver is an object that has a method that converts a list of loot conditions to a json element. 
the saver is a singleton, and i am not very sure why mojang even made the saver object, as the only used method is final and only a singleton is ever passed; the used method is only used in one place as well

Comment (Runemoro) [reply mappings/net/minecraft/predicate/entity/EntityPredicate.mapping]: > converts a list of loot conditions to a json element

> the saver is a singleton

That sounds exactly like a serializer...

Comment (liach) [reply mappings/net/minecraft/predicate/entity/EntityPredicate.mapping]: so loader/saver to deserializer/serializer?

**Review by Runemoro — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**